### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by richardm-stripe**
Codegen for openapi a8928d0.
r? @dcr-stripe
cc @stripe/api-libraries

## Changelog
* Add support for new resources `FundingInstructions` and `Terminal.Configuration`
* Add support for `create_funding_instructions` method on resource `Customer`
* Add support for `customer_balance` on `Charge.payment_method_details`, `PaymentIntent.payment_method_options`, `PaymentIntentConfirmParams.payment_method_data`, `PaymentIntentConfirmParams.payment_method_options`, `PaymentIntentCreateParams.payment_method_data`, `PaymentIntentCreateParams.payment_method_options`, `PaymentIntentUpdateParams.payment_method_data`, `PaymentIntentUpdateParams.payment_method_options`, `PaymentMethodCreateParams`, and `PaymentMethod`
* Add support for `cash_balance` on `CustomerCreateParams` and `CustomerUpdateParams`
* Add support for new value `customer_balance` on enums `CustomerListPaymentMethodsParams.type` and `PaymentMethodListParams.type`
* Add support for new value `customer_balance` on enums `PaymentIntentConfirmParams.payment_method_data.type`, `PaymentIntentCreateParams.payment_method_data.type`, and `PaymentIntentUpdateParams.payment_method_data.type`
* Add support for `amount_details` on `PaymentIntent`
* Add support for `display_bank_transfer_instructions` on `PaymentIntent.next_action`
* Add support for new value `customer_balance` on enum `PaymentMethodCreateParams.type`
* Add support for `configuration_overrides` on `Terminal.Location`, `TerminalLocationCreateParams`, and `TerminalLocationUpdateParams`

### CODE REVIEW

File: src/main/java/com/stripe/model/PaymentIntent.java [hunk 1]
```diff
@@ -1653,6 +1656,42 @@ public static class NextActionCardAwaitNotification extends StripeObject {
...
+     * Three-letter <a href="https://www.iso.org/iso-4217-currency-codes.html">ISO currency
+     * code</a>, in lowercase. Must be a <a href="https://stripe.com/docs/currencies">supported
+     * currency</a>.
+     */
+    @SerializedName("currency")
+    String currency;
+
+    /** A list of financial addresses that can be used to fund the customer balance. */
+    @SerializedName("financial_addresses")
+    List<FundingInstructions.BankTransfer.FinancialAddresses> financialAddresses;
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/model/PaymentIntent.java:1677]: Unshare, please.

File: src/main/java/com/stripe/model/terminal/Configuration.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public class Configuration extends ApiResource implements HasId {
+  @SerializedName("bbpos_wisepos_e")
+  DeviceTypeSpecificConfig bbposWiseposE;
```
Comment (dcr-stripe) [inline src/main/java/com/stripe/model/terminal/Configuration.java:28]: Let's unshare these.

File: src/main/java/com/stripe/model/terminal/Configuration.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+      this.splashscreen = new ExpandableField<File>(expandableObject.getId(), expandableObject);
+    }
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class Tipping extends StripeObject {
+    @SerializedName("aud")
+    CurrencySpecificConfig aud;
```
Comment (dcr-stripe) [inline src/main/java/com/stripe/model/terminal/Configuration.java:268]: Let's unshare these as well.

File: src/test/java/com/stripe/functional/GeneratedExamples.java [hunk 4]
```diff
@@ -3060,4 +3060,90 @@ public void testTestClockAdvance() throws StripeException {
...
+                        CustomerCreateFundingInstructionsParams.BankTransfer.RequestedAddressType
+                            .ZENGIN)
+                    .setType(
+                        CustomerCreateFundingInstructionsParams.BankTransfer.Type.JP_BANK_TRANSFER)
+                    .build())
+            .setCurrency("usd")
+            .setFundingType(CustomerCreateFundingInstructionsParams.FundingType.BANK_TRANSFER)
+            .build();
+
+    Customer customer = resource.createFundingInstructions(params);
```
Comment (dcr-stripe) [inline src/test/java/com/stripe/functional/GeneratedExamples.java:3081]: We need to update this line to be `FundingInstruction fundingInstruction = ...`

**Review by dcr-stripe — CHANGES_REQUESTED** [review]

**Comment by richardm-stripe** [comment]
re-r? @dcr-stripe

**Review by dcr-stripe — APPROVED** [review]
LGTM!

**Pull request merged**